File: Data Structures & Algorithms/is-anagram/submission-7.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        
        temp = [0] * 26
        temp2 = [0] * 26

        for ch in s:
            temp[ord(ch)-ord('a')]+=1
        
        for c in t:
            temp2[ord(c)-ord('a')]+=1
        
        print (temp)
        print (temp2)
        return temp==temp2

File: Data Structures & Algorithms/anagram-groups/submission-7.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> temp = new HashMap();

        for(int i=0; i<strs.length; i++){
            int[] keys = new int[26];
            for (int j = 0; j < strs[i].length(); j++) {
                keys[(int)(strs[i].charAt(j))-'a']+=1;
            }
            String hello = Arrays.toString(keys);
            if(!temp.containsKey(hello)) {
                temp.put(hello, new ArrayList<>());
                temp.get(hello).add(strs[i]);
            }
            else{
                temp.get(hello).add(strs[i]);
            }
        }

        
        return new ArrayList<>(temp.values());

    }
}
